src/jit/ssabuilder.cpp

```diff
@@ -855,8 +855,9 @@ void SsaBuilder::AddDefPoint(GenTree* tree, BasicBlock* blk)
     m_pCompiler->lvaTable[lclNum].lvNumSsaNames++;
 #endif
     // Record where the defn happens.
-    m_pCompiler->lvaTable[lclNum].GetPerSsaData(defSsaNum)->m_defLoc.m_blk = blk;
-    m_pCompiler->lvaTable[lclNum].GetPerSsaData(defSsaNum)->m_defLoc.m_tree = tree;
+    LclSsaVarDsc* ssaDef = m_pCompiler->lvaTable[lclNum].GetPerSsaData(defSsaNum);
+    ssaDef->m_defLoc.m_blk = blk;
+    ssaDef->m_defLoc.m_tree = tree;
```

Comment:
I considered doing that but it doesn't make any difference in terms of generated code. Besides, this `DefLoc` thing is a rather obscure type, I'm not sure why such an abstraction even exists in a codebase that looks more like C than C++.

I'd rather do something like:

```
LclSsaVarDsc* ssaDef = m_pCompiler->lvaTable[lclNum].GetPerSsaData(defSsaNum);  
ssaDef->setLocation(blk, tree);
```

but if I go down this refactoring path then I would need to change 90% of the JIT code, it's full of this type of code...

Let's see what the JIT team says.